python/rmm/rmm/_lib/memory_resource.pyx

```diff
@@ -1324,3 +1330,87 @@ def available_device_memory():
     cdef pair[size_t, size_t] res
     res = c_available_device_memory()
     return (res.first, res.second)
+
+
+cdef dict byte_sizes = {
+    "kB": 10 ** 3,
+    "MB": 10 ** 6,
+    "GB": 10 ** 9,
+    "TB": 10 ** 12,
+    "PB": 10 ** 15,
+    "KiB": 2 ** 10,
+    "MiB": 2 ** 20,
+    "GiB": 2 ** 30,
+    "TiB": 2 ** 40,
+    "PiB": 2 ** 50,
+    "B": 1,
+    "": 1,
+}
+
+byte_sizes.update({k.lower(): v for k, v in byte_sizes.items()})
+byte_sizes.update({k[0]: v for k, v in byte_sizes.items() if k and "i" not in k})
+byte_sizes.update({k[:-1]: v for k, v in byte_sizes.items() if k and "i" in k})
```

Comment:
a) Why not statically apply the updates in 1350-1352, so that:
b) Can `byte_sizes` be a local variable inside `parse_bytes` rather than a global variable?

Reply:
Also, should parse_bytes be moved to a separate file (might come in handy elsewhere).  Probably not worth an extra dependency, but this exists: https://humanfriendly.readthedocs.io/en/latest/

Reply:
I moved it to a helper file.